Backpack/Badge/Classes/Badge.swift

```diff
@@ -0,0 +1,158 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UIKit
+
+@objcMembers
+@objc
+public class BPKBadge: UIView {
+    public var message: String? {
+        get { label.text }
+        set { label.text = newValue }
+    }
+    
+    public var type: BPKBadgeType = .success {
+        didSet { updateLookAndFeel() }
+    }
+    
+    public var icon: Icon? {
+        didSet {
+            updateLookAndFeel()
+        }
+    }
+    
+    private let label: BPKLabel = {
+        let label = BPKLabel(fontStyle: .textXsEmphasized)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    private let iconView: UIImageView = {
+        let imageView = UIImageView(frame: .zero)
+        imageView.contentMode = .scaleAspectFit
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        return imageView
+    }()
+    private let containerStackView: UIStackView = {
+        let stack = UIStackView(frame: .zero)
+        stack.translatesAutoresizingMaskIntoConstraints = false
+        stack.axis = .horizontal
+        stack.spacing = BPKSpacingSm
+        return stack
+    }()
+    
+    public convenience init(type: BPKBadgeType = .success, icon: Icon? = nil, message: String) {
+        self.init(frame: .zero)
+        self.message = message
+        self.type = type
+        self.icon = icon
+        updateLookAndFeel()
+    }
+    
+    public override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+    
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        setup()
+    }
+    
+    private func updateLookAndFeel() {
+        layer.borderWidth = 0
+        backgroundColor = type.backgroundColor
+        label.textColor = type.textColor
+        
+        if type == .outline {
+            layer.borderColor = BPKColor.white.cgColor
+            layer.borderWidth = BPKBorderWidthSm
+        }
+        
+        iconView.image = icon.orNil(forType: type)
+        placeElements()
+    }
+    
+    private func placeElements() {
+        containerStackView.arrangedSubviews.forEach(containerStackView.removeArrangedSubview)
+        containerStackView.addArrangedSubview(label)
+        if let icon = icon {
+            if icon.position == .leading {
+                containerStackView.insertArrangedSubview(iconView, at: 0)
+            } else {
+                containerStackView.addArrangedSubview(iconView)
+            }
+        }
+    }
+    
+    private func setup() {
+        addSubview(containerStackView)
+        
+        layer.cornerRadius = BPKCornerRadiusXs
+        layer.masksToBounds = true
+        
+        NSLayoutConstraint.activate([
+            heightAnchor.constraint(greaterThanOrEqualToConstant: 24),
```

Comment:
Should we declare a constant for this magic number?

Reply:
I should use an existing one in fact!